src/test/java/com/snowflake/kafka/connector/records/IcebergTableStreamingRecordMapperTest.java

```diff
@@ -0,0 +1,392 @@
+package com.snowflake.kafka.connector.records;
+
+import static com.snowflake.kafka.connector.streaming.iceberg.sql.PrimitiveJsonRecord.primitiveJsonExample;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.snowflake.kafka.connector.Utils;
+import com.snowflake.kafka.connector.records.RecordService.SnowflakeTableRow;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.core.JsonProcessingException;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class IcebergTableStreamingRecordMapperTest {
```

Comment:
please think of more test cases if possible with example :)

Reply:
Looks exhaustive to me, let's wait for the bugs to happen :)